alf/algorithms/muzero_algorithm.py

```diff
@@ -321,10 +321,16 @@ def preprocess_experience(self, root_inputs: TimeStep,
             if self._reanalyze_ratio > 0:
                 # Here we assume state and info have similar name scheme.
                 mcts_state_field = 'state' + info_path[len('rollout_info'):]
-                r = torch.rand(
-                    root_inputs.step_type.shape[0]) < self._reanalyze_ratio
-                r_candidate_actions, r_candidate_action_policy, r_values = self._reanalyze(
-                    replay_buffer, env_ids[r], positions[r], mcts_state_field)
+                if self._reanalyze_ratio < 1:
+                    batch_size = root_inputs.step_type.shape[0]
+                    r = torch.randperm(
+                        batch_size) < batch_size * self._reanalyze_ratio
```

Comment:
The reason of changing from rand to randperm is to ensure that the batch size does not change, which is more memory friendly.